### DISCUSSION TITLE
Lots of writes to state.db

### DISCUSSION THREAD

**Discussion by adamfur** [k3s-io/k3s#11317] [Q&A] [answered]
**Environmental Info:**
K3s Version: 
k3s version v1.30.6+k3s1 (1829eaae)
go version go1.22.8

Node(s) CPU architecture, OS, and Version: 
Linux pi5 6.6.51+rpt-rpi-2712 #1 SMP PREEMPT Debian 1:6.6.51-1+rpt3 (2024-10-08) aarch64 GNU/Linux

Cluster Configuration:
1 server

**Describe the bug:**
# fatrace -f W
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal
k3s-server(892): W   /var/lib/rancher/k3s/server/db/state.db-wal

I see these write calls several times a second, is there anything I can do to reduce the amount? Wore out a high endurance SDcard in less than 6 months on pods that are pretty much stale.

**Comment by brandond** [comment, marked as answer]
Nope. The Kubernetes datastore is pretty chatty, and every write must be synced to the WAL in order to ensure that data is not lost in case of a power outage.
